Manual/Meta/ErrorExplanation/Example.lean

```diff
@@ -0,0 +1,244 @@
+      | HtmlT.logError "Could not find tag for error example"
+        pure .empty
+    let buttons ← titles.mapIdxM fun i (title : String) => do
+      let (tabIndex, selected) := if i == 0 then ("0", "true") else ("-1", "false")
+      let idxStr := toString i
+      return {{
+        <button type="button" role="tab"
+            aria-selected={{selected}}
+            tabindex={{tabIndex}}
+            id={{s!"{htmlId.toString}-button-{idxStr}"}}
+            aria-controls={{s!"{htmlId.toString}-panel-{idxStr}"}}>
+          {{title}}
+        </button>
+      }}
+    let panels ← contents.mapIdxM fun i b => do
+      let className := "error-example-tabpanel" ++ if i == 0 then "" else " error-example-tabpanel-hidden"
+      let idxStr := toString i
+      return {{
+        <div role="tabpanel"
+            class={{className}}
+            id={{s!"{htmlId.toString}-panel-{idxStr}"}}
+            aria-labelledby={{s!"{htmlId.toString}-button-{i}"}}>
+          {{ ← goB b }}
+        </div>
+      }}
+    pure {{
+      <div class="error-example-container">
+        <div class="error-example-tab-list" role="tablist" aria-label="Code Samples">
+          {{buttons}}
+        </div>
+        <div class="error-example-tab-view">
+          {{panels}}
+        </div>
+      </div>
+    }}
+
+
+variable [Monad m] [MonadError m]
+set_option pp.rawOnError true
+structure ErrorExampleConfig where
+  title : String
+instance : Verso.ArgParse.FromArgs ErrorExampleConfig m where
+  fromArgs :=
+    ErrorExampleConfig.mk <$> Verso.ArgParse.positional `title Verso.ArgParse.ValDesc.string
+
+/--
+Structured examples that show a minimum working example (MWE) of an error, as well as one or more
+corrections, in a fashion suitable for error explanations. The structure is relatively rigid:
+
+ * One code block labeled "broken" containing Lean code that generates an error
+ * One code block labeled "output" that contains the plan text of (one of the) generated errors
+ * One or more code blocks labeled "fixed" that contain Lean code demonstrating how the error can
+   be corrected. If there is more than one block, the block must have an additional positional
+   argument, a string describing the example. (A block labeled {lit}`fixed "descr"` will be
+   displayed as "Fixed (descr).")
+
+**Example**
+`````
+:::errorExample "Only a Dot Before the Numeral"
+```broken
+example := .3
+```
+```output
+invalid occurrence of `·` notation, it must be surrounded by parentheses (e.g. `(· + 1)`)
+```
+```fixed "make it an `Nat`"
+example := 3
+```
+```fixed "make it a `Float`"
+example := 0.3
+```
+Some explanatory text here.
+:::
+`````
+-/
+@[directive]
+def errorExample : Verso.Doc.Elab.DirectiveExpanderOf ErrorExampleConfig
+  | { title }, contents => do
+    let brokenStx :: restStx := contents.toList
+      | throwError m!"The error example had no contents"
+    let `(Lean.Doc.Syntax.codeblock|``` broken| $brokenTxt ```) := brokenStx
```

Comment:
While this totally works, it is very, very strict and doesn't participate in the usual namespacing rules.

To allow name resolution to work the right way, it might be worth instead realizing whatever the identifier the user wrote, then checking for equality with a built-in that throws errors when used outside. That'll avoid violating user expectations about names.

This is also a tiny detail and I understand if you think it's below the noise floor.